src/cryptenroll/cryptenroll-pkcs11.c

```diff
@@ -0,0 +1,97 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "cryptenroll-pkcs11.h"
+#include "hexdecoct.h"
+#include "json.h"
+#include "memory-util.h"
+#include "openssl-util.h"
+#include "pkcs11-util.h"
+#include "random-util.h"
+
+int enroll_pkcs11(
+                struct crypt_device *cd,
+                const void *volume_key,
+                size_t volume_key_size,
+                const char *uri) {
+
+        _cleanup_(erase_and_freep) void *decrypted_key = NULL, *encrypted_key = NULL;
```

Comment:
is it necessary for encrypted_key to be in erase_and_freep? What's the overhead, generally?

Reply:
no, it's not necessary, i'll drop that. 

Overhead should be irrelevant, this is not inner loop stuff, and the RSA itself is gonna by a ton slower, the explicit_bzero() that the erase stuff does should not be measurable compared to that.

I think the primary reason to drop the erase from the encrypted_key field is simply to avoid confusion: once it's encrypted it is not sensitive anymore, and the erase stuff suggests it still was. We shouldn't confuse the reader like this. Hence will remove.